### DISCUSSION TITLE
[vcpkg/scripts] WSL2 minor issue in bootstrapping script

### DISCUSSION THREAD

**Discussion by workpadwan** [microsoft/vcpkg#19822] [Q&A]
Hi,

I am trying to get vcpkg to work on WSL2 @ work and found some warnings/errors when running the script
./bootstrap-vcpkg.sh
The error I get repeats as follows:
```
./bootstrap-vcpkg.sh: 168: /path/vcpkg/scripts/bootstrap.sh: [[: not found
```

So I checked the script and it is indeed the script
scripts/bootstrap.sh at row 168

The code looks like follows there:
```
if [ "$toolData" = "" ] || [[ "$toolData" == "<?xml"* ]]; then
    echo "No entry for $toolRegexStart in $vcpkgRootDir/scripts/vcpkgTools.xml"
    return 1
fi
```

However this will not work since the script is using shell (/bin/sh) and not bash.
One "fix" could be as follows but I bet someone who knows shell better than me can make it look prettier:
```
case "$toolData" in
    "<?xml"*)
        echo "No entry for $toolRegexStart in $vcpkgRootDir/scripts/vcpkgTools.xml"
        return 1 ;;
esac
if [ "$toolData" = "" ]; then
    echo "No entry for $toolRegexStart in $vcpkgRootDir/scripts/vcpkgTools.xml"
    return 1
fi
```

This will make the warnings/errors disappear when I run the script but I will leave this as a suggestion for someone else to decide....

Cheers,

**Comment by autoantwort** [comment]
Hm seems that I have introduced that with #19640. We can do that or change the shebang to `/bin/bash`.

**Reply by workpadwan** [reply]
Hi again,

First off, many cudos for your team effort @Microsoft, this project is exactly what I need now and saves so much development time where I work!
I have no intention to nitpick here, just to be clear.

I did try to convert to /bin/bash and would just like to share some minor details about it.
I forgot to mention to replace the '==' with '=' (row 168)

When you run this script it will still give the same error.
This is because bootstrap.sh does not have the execution flag set.

When you call this from the "top-level" script bootstrap-vcpkg.sh it defaults back to shell again silently.
i.e. echo "runtime=$SHELL" in bootstrap.sh will still report /bin/bash but it will effectively adhere to /bin/sh rules.
Maybe it could be that calling a shellfile will default to sh-rules but when it's not an executable file it will ignore the shebang declaration, but I'm not sure about this.

If you for some reason want to keep /bin/sh the case ... esac above can be modifed:

```
case "$toolData" in
    "" | "<?xml"*)
        echo "No entry for $toolRegexStart in $vcpkgRootDir/scripts/vcpkgTools.xml"
        return 1 ;;
esac
```

**Reply by BillyONeal** [reply]
I think keeping `/bin/sh` is more compatible?

I wish I understood bash better. It really should be an unclosed )   ?

**Reply by talregev** [reply]
Why not switch to bash? ubuntu switch to bash and everything will follow.

**Reply by talregev** [reply]
https://stackoverflow.com/questions/3401183/bash-syntax-error-not-found

**Reply by BillyONeal** [reply]
Because @workpadwan submitted a perfectly good fix that doesn't require it

**Reply by BillyONeal** [reply]
https://github.com/microsoft/vcpkg/pull/19911/ Thanks for the fix. Do you want to submit your own PR so that you have "credit" for the merged PR since you in effect authored the fix?

**Reply by workpadwan** [reply]
@BillyONeal It's ok, just submit/commit it... I'm not that much of a poster boy.

**Reply by BillyONeal** [reply]
> @BillyONeal It's ok, just submit/commit it... I'm not that much of a poster boy.

Thanks! Landed.

**Comment by talregev** [comment]
It happen to me too. Please fix.
